### PR TITLE
C#: Add high level diagnostic messages for buildless extraction (star…

### PR DESCRIPTION
**Pull request by tamasvajk**
…t, success)

### CODE REVIEW

File: csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs [hunk 1]
```diff
@@ -310,6 +322,18 @@ void exitCallback(int ret, string msg, bool silent)
                 .Select(result => result.ToDiagnosticMessage(this, diagSeverity))
                 .ForEach(AddDiagnostic);
 
+            if (buildResult == 0 && IsBuildless)
```
Comment (igfoo) [inline csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs:325]: Should we add a `"buildless/failed"` diagnostic if we get here with `buildResult != 0`?

Comment (tamasvajk) [reply csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs:325]: If it helps the buildless telemetry dashboard, we can add it too. I think there's no such diagnostic for java at the moment.

File: csharp/extractor/Semmle.Extraction.CSharp.Standalone/Extractor.cs [hunk 2]
```diff
@@ -146,7 +146,7 @@ public static ExitCode Run(Options options)
             logger.Log(Severity.Info, "Extracting C# in buildless mode");
             using var dependencyManager = new DependencyManager(options.SrcDir, logger);
 
-            if (!dependencyManager.AllSourceFiles.Any())
+            if (!dependencyManager.NonGeneratedSourcesFiles.Any())
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp.Standalone/Extractor.cs:149]: This is a small bugfix to help write a failing buildless integration test. We shouldn't consider generated (implicit usings) files as a source file to decide on success/failure.

File: csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs]: I've added the `csharp/autobuilder/buildless/failed`.

**Review by igfoo — APPROVED** [review]

**Pull request merged**